src/struct.jl

```diff
@@ -25,58 +25,63 @@ end
 Print `x` as a struct type with mime type `MIME"text/plain"`.
 """
 function pprint_struct(io::GarishIO, mime::MIME"text/plain", @nospecialize(x))
-    t = typeof(x)
-    isstructtype(t) || throw(ArgumentError("expect a struct type, got $t"))
-    pprint_struct_type(io, mime, t); print(io.bland_io, "(")
+    if !Base.show_circular(io, x)
+        recur_io = IOContext(io.bland_io, :SHOWN_SET => x)
```

Comment:
The git diff doesn't reflect the change clearly. Actually I just added these two lines and ...